### PR TITLE
Miniz support

### PR DESCRIPTION
**Pull request by mgudemann**
this adds basic support for [miniz](https://code.google.com/archive/p/miniz) a public domain zlib compatible compression/decompression library.
Integration would remove dependency on libzip/zlib.

### CODE REVIEW

**Comment by tautschnig** [comment]
That's great! It might be better to refer to https://github.com/richgel999/miniz as code.google.com is in archive-only mode. (Not that miniz would have seen any actual updates, but at least there is active participation and forking, such as https://github.com/uroni/miniz).

**Comment by mgudemann** [comment]
thanks for the pointer, I hadn't seen Rich Geldreich's repo on github, only independent ones.

There's currently some warnings which make building with `-Werror` fail. Should we try minmizing to the bare necessities, i.e, extracting files from zip/jar?

**Comment by tautschnig** [comment]
Have you merged various improvements from the forks? The "uroni" fork mentions http://www.tenacioussoftware.com/miniz_v116_beta_r1.7z which may be important bug fixes.

On your question: If you start modifying the code, you might as well strip it down to the bare minimum.

**Comment by mgudemann** [comment]
no I have used v115_r4

the uroni fork seems interesting. On the other hand, it's MIT licensed, not
public domain, is this ok in CBMC?
​

**Comment by tautschnig** [comment]
In my understanding MIT is hardly different from public-domain, and in particular it is more liberal than BSD. So we should be good to use even the uroni fork as basis, should this be deemed useful.

**Comment by mgudemann** [comment]
ok, I'll wait for @kroening 's comment on this, whether we will do this and which version to use

the uroni patches seem to get (better?) ZIP64 support and a faster CRC32 with lookup tables. The main difference from PD to MIT is probably the obligation to include the copyright/license.
Currently MacOS builds fail (due to XXX64 calls for fseeko, stat etc.) so cannot be merged in current state anyway.

**Comment by kroening** [comment]
MIT License would be good, and this would make the build perhaps more robust.
There is also hope that this will build on Windows.
Can you please rebase?

File: src/java_bytecode/jar_file.cpp [hunk 1]
```diff
@@ -6,15 +6,13 @@ Author: Daniel Kroening, kroening@kroening.com
 
 \*******************************************************************/
 
+#include <cstring>
 #include <cassert>
+#include <iostream>
```
Comment (kroening) [inline src/java_bytecode/jar_file.cpp]: Why is iostream needed?

Comment (mgudemann) [inline src/java_bytecode/jar_file.cpp]: it's not, this was just for debugging / proof of concept, I will add the MIT-licensed "uroni" version and clean this up

**Comment by mgudemann** [comment]
cleanup on-going, windows build on Appveyor works

**Comment by mgudemann** [comment]
cleaned-up version, uses https://github.com/uroni/miniz
I ran Jamie's `sed` output script on the added files, but there's still many style errors, most are "line too long"

**Comment by mgudemann** [comment]
linter is still unhappy, but build/test on Linux/MacOS works, build on Windows as well

**Comment by kroening** [comment]
Seems ready to go after rebase.

**Comment by mgudemann** [comment]
rebase done

**Pull request merged**